firmware/application/CMakeLists.txt

```diff
@@ -492,6 +492,7 @@ add_custom_command(
 )
 
 add_executable(${PROJECT_NAME}.elf ${CSRC} ${CPPSRC} ${ASMSRC})
+set_source_files_properties(${EXTCPPSRC} PROPERTIES COMPILE_FLAGS -fno-lto)
```

Comment:
This gets more precedence over -flto as defined in cppot because it shows up later in the command line? Is this order guaranteed by cmake?

Reply:
I'm not a fan of having both --flto and --fno-lto options specified at once either.  I'm not fluent in cmake and would welcome any better ways to disable lto for just the external apps.

I have not yet found any documentation that would guarantee which will take precedence.  With the exception of the global initializers in Pacman & Tetris, it does seem to work for now, but there might be Cmake or GCC version dependencies.

Note that a WARNING message will still be displayed after the link stage if any suspect address values are found in the binary.  If we see no WARNING messages during the build, I think we can assume that the -fno-lto flag was honored.  (We could stop the make if any such messages appear, but there's a chance of false positives.)

I propose to commit this change for now to get the broken apps working, and another PR can be submitted later if a better workaround is found.

Reply:
Hi, yeah lets merge this as it is working now and we have a controlled build env! Ill try to find a safer way and submit it separately.